lib/ansible/module_utils/digital_ocean.py

```diff
@@ -0,0 +1,85 @@
+# This file is part of Ansible
+#
+# Ansible is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# Ansible is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with Ansible.  If not, see <http://www.gnu.org/licenses/>.
+#
+
+import json
+import os
+from ansible.module_utils.urls import fetch_url
+from ansible.module_utils._text import to_text
```

Comment:
We do.  I'm going to mark it stable for either 2.4 or 2.5.  The API has changed in only one subtle way in the past few releases (the default error handler changed from surrogate_or_replace to surrogate_then_replace as I found one cornercase where surrogate_or_replace could traceback).

the to_* methods make encoding issues and conversion from bytes and text much simpler than trying to ad hoc a solution.  And as a bonus, as long as it's marked unstable, I've been reviewing the code that uses it to make sure I fix them if I make any changes that would break (so far I've been lucky on that front.  the only things I've had to change were when I first introduced the functions and only a few things were using them).  Once I declare the interface stable I'll have to create new versions of the functions if I make backwards incompatible changes and then I'll rely more heavily on maintainers to port to new versions on a deprecation cycle.